### ISSUE TITLE
Useless-else-on-loop on a while loop without brake

### ISSUE THREAD

**Issue by LJBD** [PyCQA/pylint#2531]
### Steps to reproduce
1. Run `pylint` against the following code:
```python
"""Test for useless-else-on-loop.absolute_import"""

def main():
    """Main method of the test."""
    i = 0
    while i < 10:
        i += 1
    else:
        print(i)

if __name__ == "__main__":
    main()
```

### Current behavior
```test.py:8:4: W0120: Else clause on loop without a break statement (useless-else-on-loop)```

### Expected behavior
Using else on while loops doesn't require a `break` statement - else clauses are executed once the condition is false (see [docs](https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops)). There should be no warning.

### pylint --version output
```
pylint 2.1.1
astroid 2.1.0-dev
Python 3.6.4 |Anaconda, Inc.| (default, Mar 13 2018, 01:15:57) 
[GCC 7.2.0]
```

**Comment by PCManticore** [comment]
@LJBD The error is correct in this case. Your `else` clause is superfluous, because you can write it as:
```
    i = 0
    while i < 10:
        i += 1
     print(i)
```
As long as you don't have a `break` in your `while` loop, the `else` clause most of the time is just an additional artefact that can be removed.
